clients/filesystem-fuse/src/fuse_api_handle.rs

```diff
@@ -0,0 +1,437 @@
+        fh: u64,
+        flags: u32,
+        lock_owner: u64,
+        flush: bool,
+    ) -> fuse3::Result<()> {
+        self.local_fs.close_file(inode, fh).await
+    }
+
+    async fn opendir(&self, req: Request, inode: Inode, flags: u32) -> fuse3::Result<ReplyOpen> {
+        let file_handle = self.local_fs.open_dir(inode, flags).await?;
+        Ok(ReplyOpen {
+            fh: file_handle.handle_id,
+            flags: flags,
+        })
+    }
+
+    type DirEntryStream<'a>
+        = BoxStream<'a, fuse3::Result<DirectoryEntry>>
+    where
+        T: 'a;
+
+    async fn readdir<'a>(
+        &'a self,
+        req: Request,
+        parent: Inode,
+        fh: u64,
+        offset: i64,
+    ) -> fuse3::Result<ReplyDirectory<Self::DirEntryStream<'a>>> {
+        let current = self.local_fs.stat(parent).await?;
+        let files = self.local_fs.read_dir(parent).await?;
+        let entries_stream =
+            stream::iter(files.into_iter().enumerate().map(|(index, file_stat)| {
+                Ok(DirectoryEntry {
+                    inode: file_stat.inode,
+                    name: file_stat.name.clone().into(),
+                    kind: file_stat.kind,
+                    offset: (index + 3) as i64,
+                })
+            }));
+
+        let mut relative_paths = stream::iter([
+            Ok(DirectoryEntry {
+                inode: current.inode,
+                name: ".".into(),
+                kind: Directory,
+                offset: 1,
+            }),
+            Ok(DirectoryEntry {
+                inode: current.parent_inode,
+                name: "..".into(),
+                kind: Directory,
+                offset: 2,
+            }),
+        ]);
+
+        let combined_stream = relative_paths.chain(entries_stream);
+        Ok(ReplyDirectory {
+            entries: combined_stream.skip(offset as usize).boxed(),
+        })
+    }
+
+    async fn releasedir(
+        &self,
+        req: Request,
+        inode: Inode,
+        fh: u64,
+        flags: u32,
+    ) -> fuse3::Result<()> {
+        self.local_fs.close_file(inode, fh).await
+    }
+
+    async fn create(
+        &self,
+        req: Request,
+        parent: Inode,
+        name: &OsStr,
+        mode: u32,
+        flags: u32,
+    ) -> fuse3::Result<ReplyCreated> {
+        let file_handle = self
+            .local_fs
+            .create_file(parent, name.to_str().unwrap(), flags)
+            .await?;
+        Ok(ReplyCreated {
+            ttl: self.default_ttl,
+            attr: dummy_file_attr(
+                file_handle.file_id,
+                RegularFile,
+                Timestamp::from(SystemTime::now()),
+                &self.fs_context,
+            ),
+            generation: 0,
+            fh: file_handle.handle_id,
+            flags: flags,
+        })
+    }
+
+    type DirEntryPlusStream<'a>
+        = BoxStream<'a, fuse3::Result<DirectoryEntryPlus>>
+    where
+        T: 'a;
+
+    async fn readdirplus<'a>(
```

Comment:
What‘s the difference between `readdirplus` and `readdir`?
I could have a better name for `readdirplus` function.

Reply:
`readdirplus` is a new api of fuse. it has better performance